util/reconciler/defaults.go

```diff
@@ -32,3 +40,66 @@ func DefaultedLoopTimeout(timeout time.Duration) time.Duration {
 
 	return timeout
 }
+
+// getEnv return env variable
+func getEnv(key string, defaultValue string) string {
+	value := os.Getenv(key)
+	if value == "" {
+		return defaultValue
+	}
+	return value
+}
+
+// KeepRetryWithError retry based on httpResCode and httpResType.
+func KeepRetryWithError(requestStr string, httpRes *http.Response, err error, allowedErrors map[string]int) bool {
+	if err != nil {
+		buffer := new(strings.Builder)
+		_, err := io.Copy(buffer, httpRes.Body)
+		httpResBody := buffer.String()
+		httpResBodyData := []byte(httpResBody)
+		httpResBodyParsed, err := gabs.ParseJSON(httpResBodyData)
+		if err != nil {
+			return true
+		}
+		httpResCode := strings.Replace(strings.Replace(fmt.Sprintf("%v", httpResBodyParsed.Path("Errors.Code").Data()), "[", "", 1), "]", "", 1)
+		httpResType := strings.Replace(strings.Replace(fmt.Sprintf("%v", httpResBodyParsed.Path("Errors.Type").Data()), "[", "", 1), "]", "", 1)
```

Comment:
Use [this](https://github.com/outscale/osc-sdk-go/blob/v2/examples/error_test.go) instead of reading the body manually